### DISCUSSION TITLE
Thoughts on using MS4

### DISCUSSION THREAD

**Discussion by mproffitt** [musescore/MuseScore#16674] [General]
So I have been using MuseScore 4 for a few weeks now and I'd like to offer some thoughts on the program, how I found its overall usability and where I, as a composer think it could be improved.

## About my setup
Before I start I'm running MuseScore 4 as an AppImage on Ubuntu Studio 22.10 on a 32 core AMD Ryzen 9 processor with 64GB RAM.

My primary input device is currently a GAOMON graphics tablet and I use an AKAI MPK mini pro for testing chord progressions;
not really for input although occasionally I find this useful. On the output side, I use Pipewire with Carla for software based processing.

## About me
Professionally I am not really a musician anymore, I work as a solutions architect doing managed kubernetes but I am classically trained as a singer, trombonist (have played in a few orchestras and brass bands), I'm self taught on guitar (classical and electric) and piano, and was learning cello and theremin before I moved from the UK to Spain.

My background into composition started on paper, later Sibelius using versions 2 and later 3 but that was 20 years ago now.
More recently I've been using MS3 after quite a long hiatus from music caused by changing personal circumstance.

## My thoughts
So with that out of the way, I'd like to go over my thoughts and critisisms. I know for many of the critisisms there are open Github issues and some either have pull-requests or are already complete either for 4.0.2 or for 4.1.

1. I love this program. It makes composing easy. I've written more in 5 weeks using MS4 than I have in the last 5 years. Whilst not all of this can be attributed to MS4, a huge amount can be, as the result of not having to fight with the interface.

2. In MS3, it felt like everything was a pop-out window in a clunky interface. MS4 is far more compact and integrated. Huge kudos to the designers here; it feels far more professional and less of a sprawl; However:

3. I miss the tabbed interface from MS3. When I open a new score, I'd like the option of having that to be in a tab alongside rather than a whole new interface window. Yes, sometimes it's useful, particularly when transcribing between scores but at the moment, I'm looking at re-imagining some older scores and it would be nice to see these all in one place rather than needing to bounce between 4 or 5 windows.

4. Midi input. Seriously IMHO this is aweful. Notes don't play; when they do play, they occasionally don't turn off; chords occasionally arpegiate or only one note in the chord will play; pressure sensitivity is missing. I actually thought my keyboard had been damaged until I tested it in SurgeXT and found it to be working perfectly. This area needs some serious attention.

5. Note entry. I marked the score pianisimo, why is it playing fortisimo? Also, I marked Pizzicato, why am I hearing both pizzicato and arco?

6. Interface. After using it for a while, panels stop displaying and I need to restart MS4 for them to appear again. This may occasionally happen when dragging score markings from the left panel and they randomly disappear from said panel.

7. Drum entry. Lets be clear, I hated this in MS3 too so its not an MS4 issue. I work a lot with a pen, the weird key combination thing drives me nuts. I'd rather drag-drop note heads to their relevant positions. Maybe this would be different if midi input was better, then I could assign the drumpads but to be fair, I've not tested this yet (pressure sensitivity).

8. Random crashes. The one that really got me was rallentando as it took me the best part of a day to figure out what was causing it but I've also seen this with MP3 export, dynamic changes from musicXML and various other unknown reasons that happen occasionally then don't happen seem to happen again, making it hard to pinpoint what combination of factors lead into that crash.

9. The inability to assign a percussion midi channel when importing from musicXML. This couples with the incorrect playback of drum notation when imported from musicXML and copy-pasted onto a new drum line. If I rescore the drums, the position and shape is identical but the sound very different. If I copy-paste, it looks and sounds nothing alike. OK to be fair, this could be a sibelius export issue but the sum of it is that I'm left with 2 pieces with long, complex drum parts that need completely rewriting and this makes me sad.

10. Layout and cross-staff notation. It's clear some serious effort has gone into this area and this really can't be overstated however there are a few places I'd like to see improvement. Harmonics on cross-staff harp was one notable, and having the ability to change the beam length on cross-staff notation. I think the images below describe this best, from my score [Awakening 1](https://musescore.com/user/60476083/scores/9963025)

  - Cross staff beaming length (semi-quavers) - this passage really feels scrunched up to me:
![harp-layout-1](https://user-images.githubusercontent.com/1852919/223070838-cf21a058-af3f-4cd5-a8ee-ddc28a030785.png)
- Cross staff harmonics (notice how this doesn't feel as "scrunched"?)
![harp-layout-2](https://user-images.githubusercontent.com/1852919/223071010-241d7111-fd23-41c2-a7be-46fd8acb2a02.png)
- cross staff harmonics can be fixed by dragging the beams down but this is cumbersome when there are a lot of beams in a passage and didn't really translate to the web-player when I published the score.
![harp-layout-2-fixed](https://user-images.githubusercontent.com/1852919/223071207-16a673d8-f014-4b0a-850d-8ed14901281f.png)

11. Duration of breath marks. Entering a value for one basically locks the spinner so other breath marks can't be entered manually.

12. Dynamics. When I want a passage to grow, then immediately return to the previous volume there is no way to easily achieve this. having an option of "playback %" in dynamic properties would be awesome. Maybe something similar to this feature for tempo?
![tempo](https://user-images.githubusercontent.com/1852919/223074139-90a1d995-931e-491a-860c-1126b81cd1f1.png)

MuseSounds and mixer.

1. Again, I love the sound here. This completely changes the nature of how my compositions sound to me. There is far more tonal contrast which really does lead to richer sounding scores but some of that comes at a cost.

2. Some of the instruments sound unbalanced and require much vastly different volume markings or mixer changes to bring them inline with the rest of the score. Tubular bells is one in particular where I'd expect this to sound far louder. This may well be down to expectations from coming from soundfonts where everything seems to be at the same volume.

3. When scores require instruments not in MuseSounds, it requires far more effort to balance the sound. For example, I use bass guitar in a few of my compositions and if I've marked the strings mezzo-forte, bass guitar needs to be marked at most piano to sound balanced.

4. Yes, I too miss the ability to select different sounds from a soundfont but I can live without this for the moment.

5. Sometimes string entry is a little too soft or late even when accented. This is kinda obvious to me in my score [Construct and Creation](https://musescore.com/user/60476083/scores/10051915) as it can sometimes have the effect of making notes "disappear", especially on quiet, faster passages.

6. Playback - fast passages do not start cleanly, sound is delayed so half of the notes are crushed together. Doesn't happen if there is space to enter the passage (e.g. a blank bar beforehand)

7.  I invariably use the loop markers when working on a passage. When I remove them, playback regularly ignores the selected note and jumps back to the beginning of the score.

8. If I'm focusing in on the strings during playback, the page jumps to the top of the page on page turn. This makes it hard to follow what I'm doing. I would love to see this "feature" go away.

9. Lack of Jack/Pipewire support is a pain for recording.

## Overall thoughts
I've used Finale, I've used Sibelius, I've used Rosegarden, Lilypond, MS3 and none of them come close to what you're achieving here. I see comments such as "I am going back to MS3"; I won't. I love where this program is going.

Sure it has some flaws, it's missing some features and it crashes or needs to be restarted (a lot) but the benefits far outweigh the bugs and issues and I know many of these are being resolved as quickly as you are able to get to them.

I have 3 scores on musescore.com today:

* Construct and Creation (remastered from Sibelius export) https://musescore.com/user/60476083/scores/10051915
* Awakening I - A mountain sleeps. Started in MS3, finished in MS4 https://musescore.com/user/60476083/scores/9963025
* Awakening II - The forests lament https://musescore.com/user/60476083/scores/9971689

On top of this I'm working on 2 new scores, have 7 I want to remaster and a ton of inspiration for others.

I know there is some critisism here and some of it relates to issues that are already solved or being solved for a later version but I hope there is something here that can be used to improve MS4 as a product and isn't just seen as "old man shouts at cloud"

**Comment by cbjeukendrup** [comment]
Thanks for your comments! 

Many things are already on our list, to mention a few in particular:
- the tabbed interface will most likely return eventually; the current windowed interface was mainly due to technical limitations
- midi input deserves some love indeed
- percussion input will get an overhaul at some point

Some of the other things are plain bugs. It would be great if you could log these one by one at https://github.com/musescore/MuseScore/issues/new/choose. 
(That's a bit of work, sorry; but having separate issues for each bug is crucial to keep track of what was fixed already.)
Specifically: number 5, 6, 8, 9, 10, 11 from your general comments, and number 5, 7 and 8 from the playback comments. Those don't have an existing issue yet, as far as I know.

**Comment by mproffitt** [comment]
I need to confirm the second half of 5 as I have just seen similar situation inb the last 30 minutes where I pressed a midi key and got both a pizzicato and a french horn sounding, it might be that I had another window open with an active channel. For the first half of 5 I thought I saw an issue for this already but maybe it was also mentioned on the forum, I'll open a new issue about this and if it's a duplicate, we can close it.

The rallentando issue (general 8) already has an open issue from Friday (https://github.com/musescore/MuseScore/issues/16647)  for which I also have a workaround, and in fact have completed the piece I was working on when I discovered it, however the issue is very real and if a score is required for replication, I can either make Construct and Creation available in a form that forces the crash, or put something new together that replicates it.

I'll try raise for other random crashes as I am able to replicate them and provide evidence of how they are formed.

Each of the other points I will try raise as issues for but this may take me some time as I have completely different, non-musical compositions to write this week and I'd like to gather as much evidence to help you identify the issues as possible.

**Comment by mproffitt** [comment]
I've currently raised the following issues against this discussion; more of them will follow as I am able to find the time to work on replicating them.

## General points
* 5 - this is being split into two as I think this is two separate issues. The first part is raised as a feature change rather than a bug https://github.com/musescore/MuseScore/issues/16698
* 10 this again is split into two separate issues, spacing and harmonics. They are https://github.com/musescore/MuseScore/issues/16682 and https://github.com/musescore/MuseScore/issues/16697 respectively
* 8 - rallentando issue, as previously mentioned https://github.com/musescore/MuseScore/issues/16647

## Playback
* 5 - Strings playback https://github.com/musescore/MuseScore/issues/16681
* 8 - This is raised here. https://github.com/musescore/MuseScore/issues/16680

**Comment by DaddyLudwig** [comment]
2 under Playback is also something I've noticed, particularly with voices. As someone transcribing classical music into MuseScore, with lieder or really anything for Voice/Voices + Piano or small ensemble, I have to set the voice to f in dynamic and +4-6 dB in the mixer when the other instrument/instruments are mf. For something with a small orchestra + Voices like Bach's cantatas, I also have to set the voices to f in dynamic, but this time with +12 dB in the mixer, the max adjustment, again with the orchestra at mf.

With me eventually wanting to transcribe Beethoven's Ninth into MuseScore, this is kind of a big problem, cause what am I going to do for the climax where the voices and large orchestra state the Ode to Joy Theme at fortissimo? fff + an adjustment of +12 dB for the voices in the mixer? Will the voices even be heard over the orchestra with that? It's a legitimate concern to me as a classical music transcriber. And that's not to mention the lack of solo voices. But I talked to the people on Muse Hub Zendesk and they said they will eventually add solo voice sounds, so yeah, that will do more justice to operas, lieder, and pieces with both solo quartet and chorus(like Beethoven's Ninth).

Cello in piano trio is also not ideally balanced, though not nearly as bad as voice, I only have to do a +3.3 dB adjustment in the mixer and don't have to change the dynamic at all to hear the cello clearly.
